docs/blog/2025-06-20-pinned-zmk.md

```diff
@@ -0,0 +1,110 @@
+---
+title: "Pin your ZMK version"
+authors: nmunnich
+tags: [keyboards, firmware, oss, versioning]
+---
+
+ZMK is preparing for a big breaking change. As such, users should consider pinning their ZMK version. Read on to find out more about our versioning system, and how you can pin your ZMK version.
+
+<!-- truncate -->
+
+## ZMK Versioning - The 'what' and the 'why'
+
+Around 6 months ago, ZMK began using versioning. Prior to this point, all ZMK users ran their firmware off of `main`. This meant that users would always get the latest and greatest changes whenever they built new firmware, but on occasion there would be breaking changes in `main`, resulting in plenty of user confusion.
+
+Avoiding this user confusion became a limiting factor when it came to making changes. [Zephyr](https://www.zephyrproject.org/), the RTOS that ZMK is based off of, made some breaking changes in version 3.6. Hence ZMK has been stuck on Zephyr 3.5 for over 1.5 years, while Zephyr has already moved on to version 4.1. Asking when ZMK will upgrade is a frequent question we get asked, as the new version comes with plenty of improvements and drivers for newer, exciting hardware.
+
+With versioning now in place, users can pin their ZMK configuration to a particular version. This prevents sudden breaking changes from happening without user opt-in, and frees ZMK contributors to make breaking changes more frequently, as is necessary.
+
+### ZMK's Versioning System
+
+ZMK follows the [Semantic Versioning](https://semver.org/) standard. Versions are `vX.Y.Z`. ZMK uses release branches and tags: Every major or minor version bump, a new branch and tag are created. The tag points to the latest patch version. For example, when releasing `v0.2.0` the branch `v0.2-branch` and tag `v0.2` were created. When patches are made, such as for `v0.2.1`, the branch and tag are updated to point to `v0.2.1` instead of `v0.2.0`. This way, users can receive non-breaking security and bugfixes without needing to bump their version.
+
+Whenever ZMK releases a major version with breaking changes, a proper changelog will be provided detailing the major changes from the previous version, along with clear instructions on how to upgrade and fix any breaking changes.
+
+## To Pin or Not To Pin
+
+If you are a casual user of ZMK, I would always recommend pinning your version. Unpinned ZMK configurations are mainly suitable for those more heavily involved or interested in ZMK development. Moving forwards, if you are on the `main` branch of ZMK you will be considered a (highly valued) tester, rather than a user.
```

Comment:
```suggestion
For casual users of ZMK, pinning your version is recommended. Unpinned ZMK configurations are mainly suitable for those more heavily involved or interested in ZMK development. Moving forward, if you are on the `main` branch of ZMK you will be considered a (highly valued) tester, rather than a user.
```